### PR TITLE
Secp p hints

### PR DESCRIPTION
**Pull request by mmsc2**

### CODE REVIEW

File: pkg/hints/ec_hint_test.go [hunk 1]
```diff
@@ -234,3 +235,12 @@ func TestRunComputeSlopeOk(t *testing.T) {
...
 }
+
+func TestIntegrationEcDoubleSlope(t *testing.T) {
+	t.Helper()
+	cairoRunConfig := cairo_run.CairoRunConfig{DisableTracePadding: false, Layout: "all_cairo", ProofMode: false}
+	_, err := cairo_run.CairoRun("../../cairo_programs/ec_double_slope.json", cairoRunConfig)
+	if err != nil {
+		t.Errorf("TestIntegrationEcSlope failed with error:\n %v", err)
+	}
+}
```
Comment (fmoletta) [inline pkg/hints/ec_hint_test.go]: You can use the function `testProgram` here

File: pkg/hints/ec_hint.go [hunk 2]
```diff
@@ -181,3 +182,78 @@ func computeSlope(vm *VirtualMachine, execScopes ExecutionScopes, idsData IdsMan
...
+	point, err := EcPointFromVarName("point", &vm, ids_data)
+	if err != nil {
+		return err
+	}
+
+	secp_p_uncast, err := execScopes.Get("SECP_P")
+	if err != nil {
+		return err
+	}
+	secp_p := secp_p_uncast.(big.Int)
```
Comment (fmoletta) [inline pkg/hints/ec_hint.go]:
```suggestion
	secp_p, ok  := secp_p_uncast.(big.Int)
```
This cast can fail, please handle it

File: pkg/hints/ec_hint.go [hunk 3]
```diff
@@ -181,3 +182,78 @@ func computeSlope(vm *VirtualMachine, execScopes ExecutionScopes, idsData IdsMan
...
+
+	alpha_uncast, err := execScopes.Get("ALPHA")
+	if err != nil {
+		return nil
+	}
+
+	alpha := alpha_uncast.(big.Int)
```
Comment (fmoletta) [inline pkg/hints/ec_hint.go]:
```suggestion
	alpha, ok := alpha_uncast.(big.Int)
```
Handle cast failure

File: pkg/hints/field_utils.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    ids.q = q % PRIME
+%}
+*/
+
+func verifyZeroWithExternalConst(vm VirtualMachine, execScopes ExecutionScopes, idsData IdsManager) error {
+	secpPuncast, err := execScopes.Get("SECP_P")
+	if err != nil {
+		return err
+	}
+	secpP := secpPuncast.(big.Int)
```
Comment (fmoletta) [inline pkg/hints/field_utils.go]: Handle cast failure

File: pkg/hints/hint_utils/bigint_utils.go [hunk 5]
```diff
@@ -3,12 +3,51 @@ package hint_utils
...
+	res_relloc, err := ids_data.GetAddr("res", &virtual_machine)
+	if err != nil {
+		return err
+	}
+
+	value_uncast, err := exec_scopes.Get("value")
+	if err != nil {
+		return err
+	}
+	value := value_uncast.(big.Int)
```
Comment (fmoletta) [inline pkg/hints/hint_utils/bigint_utils.go]: Handle cast failure

File: pkg/hints/hint_utils/bigint_utils.go [hunk 6]
```diff
@@ -3,12 +3,51 @@ package hint_utils
...
+		return err
+	}
+	arg := make([]memory.MaybeRelocatable, 0)
+
+	for i := 0; i < 3; i++ {
+		m := memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromBigInt(&bigint3_split[i]))
+		arg = append(arg, *m)
+	}
+
+	virtual_machine.Segments.LoadData(res_relloc, &arg)
```
Comment (fmoletta) [inline pkg/hints/hint_utils/bigint_utils.go]: Handle error returned by LoadData

File: pkg/hints/hint_utils/bigint_utils.go [hunk 7]
```diff
@@ -3,12 +3,51 @@ package hint_utils
...
 	"math/big"
 
+	"github.com/lambdaclass/cairo-vm.go/pkg/lambdaworks"
 	. "github.com/lambdaclass/cairo-vm.go/pkg/lambdaworks"
+	. "github.com/lambdaclass/cairo-vm.go/pkg/types"
 	. "github.com/lambdaclass/cairo-vm.go/pkg/vm"
+	"github.com/lambdaclass/cairo-vm.go/pkg/vm/memory"
 	. "github.com/lambdaclass/cairo-vm.go/pkg/vm/memory"
 	"github.com/pkg/errors"
 )
```
Comment (fmoletta) [inline pkg/hints/hint_utils/bigint_utils.go:12]: This file (aka pkg/hints/hint_utils/bigint_utils.go) is used exclusively to support big integer struct representations used across various hints, it shouldn't contain hint implementations. Please move hint implementations to the appropiate file under the 'pkg/hints' directory

File: pkg/hints/hint_utils/bigint_utils_test.go [hunk 8]
```diff
@@ -0,0 +1,75 @@
+package hint_utils_test
+
```
Comment (fmoletta) [inline pkg/hints/hint_utils/bigint_utils_test.go]: Move this file to the hints directory & package

File: pkg/hints/hint_utils/ids_manager.go [hunk 9]
```diff
@@ -126,6 +126,7 @@ func (ids *IdsManager) GetAddr(name string, vm *VirtualMachine) (Relocatable, er
 		ids_lives := ids.GetStructField("cat", 0, vm) or ids_lives := ids.Get("cat", vm)
 		ids_paws := ids.GetStructField("cat", 1, vm)
 */
+
```
Comment (fmoletta) [inline pkg/hints/hint_utils/ids_manager.go]: why

**Review by fmoletta — COMMENTED** [review]
Please use camelCase for variable names

File: pkg/hints/bigint_utils.go [hunk 10]
```diff
@@ -1,11 +1,13 @@
-package hint_utils
+package hints
```
Comment (fmoletta) [inline pkg/hints/bigint_utils.go]: Don't move this file, it should be in the `hint_utils` package&folder

File: pkg/hints/field_utils.go [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+	v := val.Pack86()
+	q, r := v.DivMod(&v, &secpP, new(big.Int))
+
+	if r.Cmp(big.NewInt(0)) != 0 {
+		return errors.New("verify remainder is not zero: Invalid input")
+	}
+
+	quotient := memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromBigInt(q))
+	idsData.Insert("q", quotient, &vm)
+	return nil
```
Comment (fmoletta) [inline pkg/hints/field_utils.go]:
```suggestion
	return idsData.Insert("q", quotient, &vm)
```

The error returned by Insert shouldn't be ignored

**Review by fmoletta — APPROVED** [review]

**Review by jrchatruc — APPROVED** [review]

File: pkg/hints/ec_hint.go [hunk 12]
```diff
@@ -186,6 +186,7 @@ func computeSlope(vm *VirtualMachine, execScopes ExecutionScopes, idsData IdsMan
 
 /*
 Implements hint:
+<<<<<<< HEAD
```
Comment (jrchatruc) [inline pkg/hints/ec_hint.go]: Remove merge conflict marker

**Pull request merged**